Throw better errors when there are parsing issues

xavdid opened the pull request:
This is both a little more descriptive about where the error is raised, plus actually broadcasts the error contents. I added a test with a broken version of alice in wonderland (there's an invalid `&` in a chapter title)

While I was in there, I added the `files` property so we don't ship tests and the `epub` files in the npm package. Drops the package size pretty dramatically.

xavdid commented on test/test.ts:
```diff
@@ -27,4 +33,19 @@ mocha.describe('EPub', () => {
 	});
+
+	mocha.it('raises descriptive errors', () => {
+		// const epub = new EPub('./example/alice.epub')
```
what's weird is that if this line is uncommented instead of the below, the tests still pass when they shouldn't 🤔 not sure if I'm parsing it wrong or what

julien-c replied:
Hmm, mocha doesn't wait for callbacks to be executed, does it?

julien-c replied:
We should move to async/await... :)

xavdid replied:
ah, you're right. I haven't written mocha tests w/ `done` for a while.

But you're also right that we should use promises, so I added a tool for awaiting events and now it works like a charm!

xavdid commented on test/test.ts:
```diff
@@ -10,11 +10,17 @@ mocha.describe('EPub', () => {
 		const epub = new EPub('./example/alice.epub');
+
+		epub.on('end', ()=> {
+			assert.ok(epub.metadata.title)
```
added some basic coverage

xavdid commented on package.json:
```diff
@@ -19,6 +19,10 @@
     "url": "http://github.com/julien-c/epub.git"
   },
+  "files": [
+    "epub.js",
```
these are the only two files actually needed at runtime for the user

julien-c replied:
good idea

xavdid commented:
ok, we should be good here! Added extra tests. My editor also removed trailing whitespace (which makes the div look extra big, but it's not). let me know if that's an issue. :)

julien-c commented:
lgtm, feel free to merge, thanks for improving this:)

xavdid commented:
sweet! can you release to npm when you have a chance?

julien-c commented:
hmm, I think _you_ released last time, no?

julien-c commented:
(if not the case I can add you as a maintainer on npm, if that works for you)

xavdid commented:
Oh, so I am. I had forgotten. 😁 I'll take care of that!
